src/compas_fab/robots/wrench.py

```diff
@@ -0,0 +1,353 @@
+from __future__ import print_function
+
+import copy
+import numpy as np
+from scipy import stats
+
+import compas.geometry as cg
+
+__all__ = ['Wrench']
+
+
+class Wrench():
+
+	"""
+	A wrench is defined by a list of 6 float describing the 3 components of force and 3 components of torque
+
+	Parameters
+	----------
+	wrench : list of 6 floats
+		List of 6 axis forces [Fx,Fy,Fz] in Newtons. and moments [Tx,Ty,Tz] in Newton-meters
+	bias : list of 6 floats
+	mass: float [Kg]
+	center_of_mass : list of 3 float [meters]
+	"""
+
+	def __init__(self,
+				 wrench,
+				 bias = None,
+				 mass = None,
+				 center_of_mass = None):
+		
+		# wrench
+		self.wrench = wrench
+
+		# bias
+		self.bias = bias
+
+		# mass
+		if mass:
+			self.mass = mass
+		else:
+			self.mass = 8.559916898585978
+		
+		# center of mass
+		if center_of_mass:
+			self.center_of_mass = cg.Vector(center_of_mass)
+		else:
+			self.center_of_mass = cg.Vector(-0.03397799730572294, 0.042647988599212705, 0.2900206244556921)
+
+		# gravity
+		self.gravity = 9.80665	# [m/s**2]
+		self.gravity_vector_WCS = cg.Vector(0.0, 0.0, -self.gravity)
+	
+	# ==========================================================================
+	# factory
+	# ==========================================================================
+
+	@classmethod
+	def from_msg(cls, msg, bias = None, mass = None, center_of_mass = None):
+		"""Construct a wrench from its ROS message representation.
+		http://docs.ros.org/kinetic/api/geometry_msgs/html/msg/WrenchStamped.html
+
+		Parameters
+		----------
+		data : :obj:`dict`
+			The data dictionary.
+
+		Returns
+		-------
+		Wrench
+			The constructed wrench.
+		
+		Note:
+			Used from creating Wrench from ROS message
+		"""
+		wrench = [
+			msg["wrench"]["force"]["x"],
+			msg["wrench"]["force"]["y"],
+			msg["wrench"]["force"]["z"],
+			msg["wrench"]["torque"]["x"],
+			msg["wrench"]["torque"]["y"],
+			msg["wrench"]["torque"]["z"]
+		]
+		return cls(wrench, bias = bias, mass = mass, center_of_mass = center_of_mass)
+	
+	@classmethod
+	def by_samples(cls, wrenches, bias = None, mass = None, center_of_mass = None, proportiontocut = 0.0):
```

Comment:
not sure when this is needed.. Example?
also: isnt' this again rather a `ForceTorqueSensor` method?

Reply:
I used this method to quickly filter the force/torque data that I read from the force torque sensor. I added the example